web/client/components/geostory/contents/EditorCustomizations/CustomEditorLink.jsx

```diff
@@ -0,0 +1,411 @@
+                className="rdw-link-modal-target-option"
+                htmlFor="openLinkInNewWindow"
+            >
+                <input
+                    id="openLinkInNewWindow"
+                    type="checkbox"
+                    defaultChecked={linkTargetOption === '_blank'}
+                    value="_blank"
+                    onChange={this.updateTargetOption}
+                />
+                <span>
+                    {translations['components.controls.link.linkTargetOption']}
+                </span>
+            </label>
+          </>
+              )}
+              <span className="rdw-link-modal-buttonsection">
+                  <button
+                      className="rdw-link-modal-btn"
+                      onClick={this.addLink}
+                      disabled={!linkTarget || !linkTitle}
+                  >
+                      {translations['generic.add']}
+                  </button>
+                  <button className="rdw-link-modal-btn" onClick={doCollapse}>
+                      {translations['generic.cancel']}
+                  </button>
+              </span>
+          </div>
+      );
+  }
+
+  renderInFlatList() {
+      const {
+          config: { options, link, unlink, className },
+          currentState,
+          expanded,
+          translations
+      } = this.props;
+      const { showModal } = this.state;
+      return (
+          <div
+              className={classNames('rdw-link-wrapper', className)}
+              aria-label="rdw-link-control"
+          >
+              {options.indexOf('link') >= 0 && (
+                  <Option
+                      value="unordered-list-item"
+                      className={classNames(link.className)}
+                      onClick={this.signalExpandShowModal}
+                      aria-haspopup="true"
+                      aria-expanded={showModal}
+                      title={link.title || translations['components.controls.link.link']}
+                  >
+                      <img src={link.icon} alt="" />
+                  </Option>
+              )}
+              {options.indexOf('unlink') >= 0 && (
+                  <Option
+                      disabled={!currentState.link}
+                      value="ordered-list-item"
+                      className={classNames(unlink.className)}
+                      onClick={this.removeLink}
+                      title={
+                          unlink.title || translations['components.controls.link.unlink']
+                      }
+                  >
+                      <img src={unlink.icon} alt="" />
+                  </Option>
+              )}
+              {expanded && showModal ? this.renderAddLinkModal() : undefined}
+          </div>
+      );
+  }
+
+  renderInDropDown() {
+      const {
+          expanded,
+          onExpandEvent,
+          doCollapse,
+          doExpand,
+          onChange,
+          config,
+          currentState,
```

Comment:
The implementations is working but there are some small issue with the UI:

- [ ] the dropdown opens in the modal so part of the options are not visible. We could use an overflow visible to the link modal and set a fixed height that could contain the configuration with external link and section link
![image](https://user-images.githubusercontent.com/19175505/93438552-beb8fd00-f8cd-11ea-82d8-3dc74786b82c.png)

- [ ]  after selecting a section the id is displayed instead of the section name. Could be possible to use react-select inside the MSLinkOptions? so we can pass an object value and array of object options { value:  , label:  }

![wrong-name](https://user-images.githubusercontent.com/19175505/93439401-cd53e400-f8ce-11ea-9415-1c8cb62a5940.gif)

- [ ] when I try to update an existing link, title and target are not recognized by the select input but it shows the default placeholder

![recognize](https://user-images.githubusercontent.com/19175505/93442130-3a1bae00-f8d0-11ea-8cfe-adf2bba7939b.gif)

Note: to be able to solve this last issue you need the attributes property in the currentState. (see https://github.com/geosolutions-it/react-draft-wysiwyg/pull/4#pullrequestreview-490340402).

Reply:
@AGMETEOR please use lower case for folder names -> `editorcustomizations/CustomEditorLink.jsx`